vm/vm.go

```diff
@@ -288,6 +328,46 @@ func (vm *Engine) validateStateTransition(virtualTx *wire.MsgTx) error {
 				spew.Sdump(witness.PrevID))
 		}
 
+		// Assume the default flag of SigHashAll.
+		sighashFlag := extractSighashFromWitnessStack(witness.TxWitness)
+
+		// Check that we support the provided sighash flag.
+		switch sighashFlag {
+		case txscript.SigHashDefault, txscript.SigHashAll,
```

Comment:
Why do we need to officially support some flags vs others? Stuff like single or anyonecan pay has no bearing at this level. It's also the case that the logic above doesn't _exhaustively_ extract sighashes: it assumes keyspend only. All sigs within the witness can _also_ have sighashes, eg: someone doing `OP_CHECKSGIADD`.

Reply:
Also sighash values are masks, you can't just compare directly.

Reply:
> Stuff like single or anyonecan pay has no bearing at this level

So should I allow them even though they'll have no effect, or a malformed effect?